Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,277 @@
+    /// HTML into a string
+    func getHTML(block: @escaping (String) -> ()){
+        evaluateJavaScript("document.documentElement.outerHTML.toString()",
+                           completionHandler: { (html: Any?, error: Error?) in
+                            if let html = html {
+                                block(html as! String)
+                            }
+        })
+    }
+
+    /// Loads login web page
+    func handleLogin() {
+        loginCount = 0
+        stage = .loginScreen
+
+        // Remove cache
+        URLCache.shared.removeAllCachedResponses()
+
+        if let loginURL = loginURL {
+            load(URLRequest(url: loginURL))
+        }
+    }
+    
+    func failedToLogin() -> Bool {
+        return loginCount > 1
+    }
+    
+    @objc func login() {
+        let javascript = "document.getElementsByName('netid')[0].value = '\(netid)';document.getElementsByName('password')[0].value = '\(password)';document.forms[0].submit();"
+        
+        evaluateJavaScript(javascript) { (result: Any?, error: Error?) -> Void in
+            if let error = error {
+                print(error)
+                self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: error.localizedDescription)
+            } else {
+                if self.failedToLogin() {
+                    self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: "Incorrect netid and/or password")
+                }
+            }
+            self.loginCount += 1
+        }
+    }
+    
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        self.getStageAndRunBlock {
+            switch self.stage {
+            case .loginFailed:
+                self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: "Incorrect netid and/or password")
+            case .loginScreen:
+                if self.loginCount < 1 {
+                    self.login()
+                }
+            case .finished(let sessionId):
+                self.delegate?.brbConnectionHandlerDelegateDidRetrieve(sessionID: sessionId)
+            default: break
+            }
+        }
+    }
+    
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: error.localizedDescription)
+    }
+
+    /// Gets the stage enum for the currently displayed web page and runs a
+    /// block after fetching the HTML for the page. Does not guarantee
+    /// Javascript will finish running before the block is executed.
+    func getStageAndRunBlock(block: @escaping () -> ()) {
+        getHTML(block: { (html: String) -> () in
+            if self.failedToLogin() {
+                self.stage = .loginFailed
+            } else if html.contains("<h2>Login OK</h2>") {
+                self.stage = .transition
+            } else if html.contains("<h1>CUWebLogin</h1>") {
```

Comment:
i'm curious if there is a safer way to do this that doesn't involve checking the text field exactly (what if they change it to "CU Web Login" in the future?)